### ISSUE TITLE
Problem Matcher annotations do not appear inline

### ISSUE THREAD

**Issue by kimburgess** [actions/toolkit#810]
**Describe the bug**
When log output is detected via a [problem matcher](https://github.com/actions/toolkit/blob/master/docs/problem-matchers.md), annotations do not appear to attach inline when viewed from github.com. These _do_ however appear in responses from api.github.com.

**To Reproduce**
https://github.com/kimburgess/docs/runs/2597266051
https://api.github.com/repos/kimburgess/docs/check-runs/2597266051/annotations
Note that annotations do show on the [associated summary](https://github.com/kimburgess/docs/actions/runs/848405643), however do not appear linked to files within the repo.

**Expected behavior**
Annotations created via problem matchers should be visible in the same way that annotations created via direct API interaction appear (e.g. https://github.com/PlaceOS/docs/runs/2596944916)

**Comment by thboop** [comment]
Hey @kimburgess can you try [enabling debug logging](https://github.com/actions/toolkit/blob/main/docs/problem-matchers.md#file-property-getting-dropped) to troubleshoot why the file path may be getting dropped to help us troubleshoot further?

**Comment by kimburgess** [comment]
@thboop apologies - the above links would have been nuked as I had deleted that repo (was a test fork for repro).

The problem matcher does work does correctly detect and annotation the issues (confirmed not being dropped as they would also appear on the API endpoint for the check run). I have just recreated that test repo, and in the process think I may have found the original cause of the issue—human error / confusion (and possibly a UI omission).

Annotations _do_ show within a PR context. For example: https://github.com/kimburgess/docs/pull/2/files.

Similarly they also show directly on commits: https://github.com/kimburgess/docs/commit/3b2f00ccbea5a82f80af9a9f3c902652a899266f

Their context is just _not_ visible from the `Actions` view: https://github.com/kimburgess/docs/actions/runs/892718805.

This threw me as annotations associated with manually created checks do show with that context. Example: https://github.com/PlaceOS/docs/runs/2596944916.

I'll keep the above test repo available for the next 2 weeks in case it is of use / interest, but will close this issue with this behaviour now known.
